modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -1103,19 +1109,33 @@ private void replaceCallWithConstant(
         Object receiver
     ) {
         Object[] args = new Object[irInvokeCallMemberNode.getArgumentNodes().size()];
+
+        List<Tuple<Integer,String>> argumentsShouldBeConstants  = new ArrayList<>();
         for (int i = 0; i < irInvokeCallMemberNode.getArgumentNodes().size(); i++) {
             ExpressionNode argNode = irInvokeCallMemberNode.getArgumentNodes().get(i);
             IRDConstant constantDecoration = argNode.getDecoration(IRDConstant.class);
             if (constantDecoration == null) {
-                // TODO find a better string to output
-                throw irInvokeCallMemberNode.getLocation()
-                    .createError(
-                        new IllegalArgumentException(
-                            "all arguments to [" + javaMethod.getName() + "] must be constant but the [" + (i + 1) + "] argument isn't"
-                        )
-                    );
+                // offering the symbol name in error message (CastNode was evolved from ESymbol)
+                String argumentName = argNode instanceof CastNode
+                    ? ((CastNode) argNode).getChildNode().getDecoration(IRDName.class).getValue()
+                    : "";
+                argumentsShouldBeConstants.add(Tuple.tuple(i, argumentName));
+            } else {
+                args[i] = constantDecoration.getValue();
             }
-            args[i] = constantDecoration.getValue();
+        }
+        if (argumentsShouldBeConstants.size() > 0 ){
+            throw irInvokeCallMemberNode
+                .getLocation()
+                .createError(new IllegalArgumentException(String.format(
+                    Locale.ROOT,
+                    "All arguments of the [%s] method must be constants, but the following arguments are not: %s",
+                    javaMethod.getName(),
+                    argumentsShouldBeConstants
```

Comment:
The stream and `Tuple` use is unnecessary, why not build the error directly?

Reply:
My willingness is to print out all the errors at once. I will rollback this.